pandas/core/series.py

```diff
@@ -198,18 +198,59 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
                     data = data.reindex(index, copy=copy)
                 data = data._data
             elif isinstance(data, dict):
-                if index is None:
-                    if isinstance(data, OrderedDict):
-                        index = Index(data)
+                if data:
+                    keys, values = zip(*compat.iteritems(data))
+                    # Workaround for #18485 - part 1/3
+                    maybe_mi_keys = Index(list(keys), tupleize_cols=True)
+                    if isinstance(maybe_mi_keys, MultiIndex):
+                        remap_to_mi = True
+                        keys = Index(list(keys), tupleize_cols=False)
                     else:
-                        index = Index(_try_sort(data))
-
-                try:
-                    data = index._get_values_from_dict(data)
-                except TypeError:
-                    data = ([data.get(i, np.nan) for i in index]
-                            if data else np.nan)
-
+                        remap_to_mi = False
+                        keys = maybe_mi_keys
```

Comment:
this is WAY too complicated, likely slow. pls try to simplify. at the very least, should all be split out to a more accessible location, e.g. move to a helper function, below _sanitize_array is prob ok. call it _dict_to_array or something.

Reply:
going to need quite some simplification. I am ok with throwing out some cases if that will help.